src/Tokenizer/Tokens.php

```diff
@@ -293,6 +293,9 @@ public function offsetUnset($index): void
      */
     public function offsetSet($index, $newval): void
     {
+        $this->blockStartCache = [];
```

Comment:
In the case of changing braces from on type of another this still needs to clear the cache, for example from `$a[0]` to `$a{0}`. One could also make the brace type part of the cache "key" so this cache clearing here would not be needed indeed.